### ISSUE TITLE
Getting accessibility alerts as 'Missing fieldset' and 'Broken same-page link' on 'Products' page.

### ISSUE THREAD

**Issue by Shalini-git25** [DFE-Digital/find-information-products-services-frontend#104]
**Describe the bug**
When using WAVE evaluation tool for accessibility testing on 'Search and filter products and services' page, getting alerts as below, which seems to be issues -
1. 3 X Missing fieldset
2. 2 X Broken same-page link

**To Reproduce**
Steps to reproduce the behaviour:
1. Click on any sub categories link such as 'Email' for Channel category.
2. User is navigated to 'Search and filter products and services' page.
3. Using WAVE tool for accessibility checks on this page.
4. Multiple alerts are displayed in which 2 alerts seems to be issue. Below are details - 
a. 3 X Missing fieldset - pointing to 'Request', 'Chat' and 'API' checkboxes fields.
b. 2 X Broken same-page link - it's hidden on the page.

**Expected behavior**
Any accessibility alerts which seems to be issue should be fixed and no errors should be present on the Products page.

**Screenshots**
1. 3 X Missing fieldset

<img width="1114" height="666" alt="Image" src="https://github.com/user-attachments/assets/e8bbe2ae-29ee-491b-a7a1-f11f20aaaa35" />

2. 2 broken same page link 

<img width="2237" height="1361" alt="Image" src="https://github.com/user-attachments/assets/77fbc6e0-6f30-46a2-9b0c-2c2a7af5eeec" />

**Additional context**
The accessibility alert - '2 X Broken same-page link' is present on all the pages of application.
